Fix for darwin build.

hzeller opened the pull request:
Related: #199919 as this is fixing the Darwin build.

###### Description of changes

Update Surelog/UHDM to latest 1.45, which uses the googletest version that can deal with temporary directories on Darwin (the current 1.12.x  [works with environment variables](https://github.com/google/googletest/commit/5e6a533680fc8292c31f31664d80c48440d4a526) on _all_ platforms to determine temp-dir, while previous versions did not and, on Darwin, just assumed `/tmp` which did not work in the nix build environment)

Update corresponding f4pga plugin.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s) (using **ofborg**)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable (check phase of affected packages on all platforms ran)
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

hzeller committed:
uhdm: 0.9.1.40 -> 1.45

Update UHDM and also fix for Darwin build.
The gtest version used in the project unconditionally uses
/tmp as temporary directory in Darwin and does not honor the
environment variables introduced in gtest 1.12.
Use that gtest version so that it will work in the Hydra
build environment.

Related: #199919 as this is fixing the Darwin build.

Signed-off-by: Henner Zeller <h.zeller@acm.org>

hzeller commented:
@ofborg build uhdm surelog yosys-symbiflow

hzeller commented:
@ofborg build uhdm surelog yosys-symbiflow

hzeller commented:
@ofborg build uhdm surelog yosys-symbiflow

hzeller commented:
Also adding @Artturin as we were working on Verible earlier to get it to compile on all platforms.
Here is another fix in a different corner of the packages towards a happy hydra.

Artturin commented:
These should be in separate commits

> ofborg build uhdm surelog yosys-symbiflow

Ofborg will automatically build the package as long as the commit message follows the format `attr: message`

hzeller committed:
surelog: 1.40 -> 1.45

Update Surelog (going along with uhdm 1.45). Fix flaky tests
to to parallel execution of ctest by patching in a fix that is
already upstream.

Result zero hydra failures #199919

Also: using flatbuffers from nix environment instead of third_party/

Signed-off-by: Henner Zeller <h.zeller@acm.org>

hzeller committed:
yosys-symbiflow: 2022.09.27 -> 2022.11.07

Update plugins to be compatible with latest surelog/uhdm (1.45).
Also fix expected googletest path so that it comes from the
nix environment, not third_party/.

Signed-off-by: Henner Zeller <h.zeller@acm.org>

Atemu commented on pkgs/applications/science/logic/surelog/default.nix:
```diff
@@ -34,8 +44,14 @@ stdenv.mkDerivation rec {
+    "-DSURELOG_USE_HOST_FLATBUFFERS=On"
+  ];
+
+
```
Nit:

```suggestion
```

hzeller replied:
Done.

Atemu commented on pkgs/applications/science/logic/surelog/default.nix:
```diff
@@ -7,20 +7,30 @@
     fetchSubmodules = true;
   };
 
+  patches = [
+    (fetchpatch {
+      url = "https://github.com/chipsalliance/Surelog/commit/9a54efbd156becf65311a4272104810f36041fa6.patch";
+      sha256 = "sha256-rU1Z/0wlVTgnPLqTN/87n+gI1iJ+6k/+sunVVd0ulhQ=";
+      name = "parallel-test-running.patch";
+    })
+  ];
```
Why do we need this? When can we remove this again?

A comment would be helpful.

hzeller replied:
It prevents race conditions in tests. These show up as we run `ctest` with parallel execution enabled.

It can be removed with the next tagged version of this project.

hzeller replied:
Added comment.

Atemu commented on pkgs/applications/science/logic/uhdm/default.nix:
```diff
@@ -3,26 +3,41 @@
     fetchSubmodules = true;
   };
 
+  patches = [
+    (fetchpatch {
+      url = "https://github.com/chipsalliance/UHDM/commit/ad60fdb65a7c49fdc8ee3fffdca791f9364af4f5.patch";
+      sha256 = "sha256-IkwnepWWmBychJ0mu+kaddUEc9jkldIRq+GyJkhrO8A=";
+      name = "allow-local-gtest.patch";
+    })
+  ];
+
```
Why do we need this? When can we remove this again?

A comment would be helpful.

hzeller replied:
This provides the ability to use googletest on the system itself (so that we can use `-DUHDM_USΕ_HOST_GTEST` in the cmake options).
It prevents using the googletest that otherwise comes as submodule of third_party/.

It can be removed with the next tagged version of this project.

Atemu commented on pkgs/applications/science/logic/uhdm/default.nix:
```diff
@@ -33,12 +48,6 @@ stdenv.mkDerivation rec {
     rm -rf $out/lib/uhdm
   '';
 
-  prePatch = ''
-    substituteInPlace CMakeLists.txt --replace \
-    'capnp compile' \
-    'capnp compile --src-prefix=''${GENDIR}/..'
-  '';
-
```
Why is this not necessary anymore?

Atemu reviewed (commented):
Result of `nixpkgs-review pr 200145` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>uhdm</li>
  </ul>
</details>

hzeller commented:
@Artturin : Split into three commits, one for each project
@Atemu : Addressed review comments and added comments in the patch sections.
